src/kokkos_abstraction.hpp

```diff
@@ -62,16 +63,21 @@ template <typename T>
 using ParArray1D = Kokkos::View<T *, LayoutWrapper, DevMemSpace>;
 template <typename T>
 using ParArray2D = Kokkos::View<T **, LayoutWrapper, DevMemSpace>;
```

Comment:
It works fine if I wrap `ParArray2D` as well with `ParArrayGeneric` but wrapping `ParArray1D` causes a number of compilation errors. I think this is because `ParArray1D`s are being used as if they are raw `Kokkos::views` in many places throughout the code and I have not added every possible overload for `ParArrayGeneric` in the overloads I put in the `Kokkos` namespace.

There is no fundamental reason why they shouldn't all be specializations of `ParArrayGeneric`. I was mostly just being lazy. Next week when I am not burned out on looking at compiler error explosions I will try to make `ParArray1D` work as well.

Reply:
done.